src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/AggregationIT.java

```diff
@@ -426,59 +426,61 @@ public void termsWithOrder() throws Exception {
-        JSONArray buckets = ageAgg.getJSONArray("buckets");
-
-        int previousBucketCount = 0;
-        int currentBucketCount;
-        for (int i = 0; i < buckets.length(); ++i) {
-            currentBucketCount = (int) buckets.query(String.format(Locale.ROOT, "/%d/COUNT(*)/value", i));
-
-            if (0 == i) {
-                previousBucketCount = currentBucketCount;
-                continue;
-            }
+    public void orderByAscTest() {
+        JSONObject response = executeJdbcRequest(String.format("SELECT COUNT(*) FROM %s " +
+                "GROUP BY gender ORDER BY COUNT(*)", TEST_INDEX_ACCOUNT));
+
+        verifySchema(response, schema("COUNT(*)", null, "integer"));
+        verifyDataRows(response,
+                rows(493),
+                rows(507));
+    }
 
-            Assert.assertThat(currentBucketCount, greaterThanOrEqualTo(previousBucketCount));
-            previousBucketCount = currentBucketCount;
-        }
+    @Test
+    public void orderByAliasAscTest() {
+        JSONObject response = executeJdbcRequest(String.format("SELECT COUNT(*) as count FROM %s " +
+                "GROUP BY gender ORDER BY count", TEST_INDEX_ACCOUNT));
+
+        verifySchema(response, schema("count", "count", "integer"));
+        verifyDataRows(response,
+                rows(493),
+                rows(507));
     }
 
     @Test
     public void orderByDescTest() throws IOException {
+        JSONObject response = executeJdbcRequest(String.format("SELECT COUNT(*) FROM %s " +
+                "GROUP BY gender ORDER BY COUNT(*) DESC", TEST_INDEX_ACCOUNT));
 
-        JSONObject result = executeQuery(String.format("SELECT COUNT(*) FROM %s GROUP BY age" +
-                " ORDER BY COUNT(*) DESC", TEST_INDEX_ACCOUNT));
-        JSONObject ageAgg = getAggregation(result, "age");
-        JSONArray buckets = ageAgg.getJSONArray("buckets");
-
-        int previousBucketCount = 0;
-        int currentBucketCount;
-        for (int i = 0; i < buckets.length(); ++i) {
-            currentBucketCount = (int) buckets.query(String.format(Locale.ROOT, "/%d/COUNT(*)/value", i));
-
-            if (0 == i) {
-                previousBucketCount = currentBucketCount;
-                continue;
-            }
+        verifySchema(response, schema("COUNT(*)", null, "integer"));
+        verifyDataRows(response,
+                rows(493),
+                rows(507));
+    }
 
-            Assert.assertThat(currentBucketCount, lessThanOrEqualTo(previousBucketCount));
-            previousBucketCount = currentBucketCount;
-        }
+    @Test
+    public void orderByAliasDescTest() throws IOException {
+        JSONObject response = executeJdbcRequest(String.format("SELECT COUNT(*) as count FROM %s " +
+                "GROUP BY gender ORDER BY count DESC", TEST_INDEX_ACCOUNT));
+
+        verifySchema(response, schema("count", "count", "integer"));
+        verifyDataRows(response,
+                rows(493),
+                rows(507));
```

Comment:
So this one is testing `ORDER BY count DESC`, should we assert the rows in reversed order?

Probably we need to replace the Hamcrest matcher `containsInAnyOrder`:

```
    @SuppressWarnings("unchecked")
    public static void verifyDataRows(JSONObject response, Matcher<JSONArray>... matchers) {
        verify(response.getJSONArray("datarows"), matchers);
    }

    @SuppressWarnings("unchecked")
    public static <T> void verify(JSONArray array, Matcher<T>... matchers) {
        List<T> objects = new ArrayList<>();
        array.iterator().forEachRemaining(o -> objects.add((T) o));
        assertEquals(matchers.length, objects.size());
        assertThat(objects, containsInAnyOrder(matchers));
    }
```

Reply:
sure. done.